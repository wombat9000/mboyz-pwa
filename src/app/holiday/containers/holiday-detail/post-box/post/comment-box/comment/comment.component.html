<div class="comment-body" *ngIf="user$ | async as user">
  <span class="author">{{user.displayName}}</span>
  <span class="message">{{comment.text}}</span>
</div>
<div class="info">
  <span class="created-text">{{formatDate(comment.created)}} um {{formatTime(comment.created)}}</span>
</div>
