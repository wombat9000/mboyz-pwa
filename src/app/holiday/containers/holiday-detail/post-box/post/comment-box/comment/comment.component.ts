import {Component, Input, OnInit} from '@angular/core';
import {Observable} from 'rxjs/Observable';
import * as moment from 'moment';
import {User} from '../../../../../../../auth/services/auth.service';
import {UserFirestore} from '../../../../../../../auth/services/user-firestore.service';
import {Comment} from '../../../../../../services/comment-firestore.service';

@Component({
  selector: 'app-comment',
  templateUrl: './comment.component.html',
  styleUrls: ['./comment.component.scss']
})
export class CommentComponent implements OnInit {

  @Input()
  comment: Comment;

  user$: Observable<User>;

  constructor(private userFS: UserFirestore) {
  }

  ngOnInit() {
    this.user$ = this.userFS.observeById(this.comment.authorId);
  }

  formatDate(isoString: string): string {
    return moment(isoString).format('Do MMMM');
  }

  formatTime(isoString: string): string {
    return moment(isoString).format('LT');
  }
}
