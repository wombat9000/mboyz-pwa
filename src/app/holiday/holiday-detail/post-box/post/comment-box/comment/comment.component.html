<div class="comment-body" *ngIf="user | async as user2">
  <span class="author">{{user2.displayName}}</span>
  <span class="message">{{comment.text}}</span>
</div>
<div class="info">
  <span class="created-text">{{comment.created.format('Do MMMM')}} um {{comment.created.format('LT')}}</span>
</div>
