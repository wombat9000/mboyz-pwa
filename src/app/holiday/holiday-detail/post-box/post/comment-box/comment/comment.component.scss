@import '~@angular/material/theming';
@import '../../../../../../../mboyz-theme';

:host {
  .author {
    color: mat-color($mboyz-primary);
    font-weight: bold;
  }

  .info {
    font-size: 80%;
    color: lighten(black, 50%);
    margin-top: 2px;
  }
}
