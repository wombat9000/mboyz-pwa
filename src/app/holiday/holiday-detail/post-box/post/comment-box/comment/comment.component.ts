import {Component, Input, OnInit} from '@angular/core';
import {UserFirestore} from '../../../../../../core/user-firestore.service';
import {User} from '../../../../../../core/auth.service';
import {Observable} from 'rxjs/Observable';
import {Comment} from '../comment-firestore.service';
import moment = require('moment');

@Component({
  selector: 'app-comment',
  templateUrl: './comment.component.html',
  styleUrls: ['./comment.component.scss']
})
export class CommentComponent implements OnInit {

  @Input()
  comment: Comment;

  user$: Observable<User>;

  constructor(private userFS: UserFirestore) {
  }

  ngOnInit() {
    this.user$ = this.userFS.observeById(this.comment.authorId);
  }

  formatDate(isoString: string): string {
    return moment(isoString).format('Do MMMM');
  }

  formatTime(isoString: string): string {
    return moment(isoString).format('LT');
  }
}
