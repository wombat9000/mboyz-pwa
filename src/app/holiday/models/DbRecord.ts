export interface DbRecord {
  readonly id: string
}
