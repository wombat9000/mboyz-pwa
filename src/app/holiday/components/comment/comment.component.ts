import {Component, Input, OnInit} from '@angular/core';
import * as moment from 'moment';
import {MtravelUser} from '../../../auth/services/auth.service';
import {CommentDTO} from '../../models/comment';
import {UserService} from '../../../auth/services/user.service';
import {Observable} from 'rxjs/index';

@Component({
  selector: 'app-comment',
  template: `
    <div class="comment-body" *ngIf="user$ | async as user">
      <span class="author">{{user.displayName}} </span>
      <span class="message">{{text}}</span>
    </div>
    <div class="info">
      <span class="created-text">{{date}} um {{time}}</span>
    </div>
  `,
  styleUrls: ['./comment.component.scss']
})
export class CommentComponent implements OnInit {

  @Input()
  comment: CommentDTO;

  user$: Observable<MtravelUser | undefined>;

  constructor(private userFS: UserService) {
  }

  get text() {
    return this.comment.text;
  }

  get date() {
    return moment(this.comment.created).format('Do MMMM');
  }

  get time() {
    return moment(this.comment.created).format('LT');
  }

  ngOnInit() {
    this.user$ = this.userFS.observeById(this.comment.authorId);
  }
}
