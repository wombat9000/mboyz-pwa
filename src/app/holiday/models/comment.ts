import {DbRecord} from './DbRecord';

export interface CommentDTO extends DbRecord {
  readonly text: string;
  readonly postId: string;
  readonly holidayId: string;
  readonly authorId: string;
  readonly created: string;
}

export function asComment(data: any): CommentDTO {
  return {
    id: data.id,
    text: data.text,
    authorId: data.authorId,
    postId: data.postId,
    holidayId: data.holidayId,
    created: data.created
  };
}

export function newTestComment(id: string): CommentDTO {
  return {
    id: id,
    text: 'someText',
    authorId: 'someAuthor',
    postId: 'somePostId',
    holidayId: 'someHolidayId',
    created: 'someCreatedDate'
  };
}
