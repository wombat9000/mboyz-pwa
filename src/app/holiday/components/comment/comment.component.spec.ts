import {async, ComponentFixture, TestBed} from '@angular/core/testing';

import {CommentComponent} from './comment.component';
import {DebugElement} from '@angular/core';
import {MtravelUser} from '../../../auth/services/auth.service';
import {CommentDTO} from '../../models/comment';
import moment = require('moment');

describe('CommentComponent', () => {
  let component: CommentComponent;
  let debugElement: DebugElement;
  let fixture: ComponentFixture<CommentComponent>;

  const someUser: MtravelUser = {
    id: 'someAuthorId',
    displayName: 'someDisplayName',
    photoURL: 'somePhotoUrl',
    email: 'someEmail'
  };

  const someComment: CommentDTO = {
    id: 'someId',
    postId: 'somePost',
    holidayId: 'someHolidayId',
    authorId: 'someAuthor',
    text: 'someMessage',
    created: moment('2016-01-01').toISOString(),
  };

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [CommentComponent]
    })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CommentComponent);
    debugElement = fixture.debugElement;
    component = fixture.componentInstance;
    component.comment = someComment;
    component.author = someUser;
    fixture.detectChanges();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
