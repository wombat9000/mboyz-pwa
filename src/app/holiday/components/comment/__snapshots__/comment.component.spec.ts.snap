// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentComponent should match snapshot 1`] = `
<app-comment
  author={[Function Object]}
  comment={[Function Object]}
>
  <div
    class="comment-body"
  >
    <span
      class="author"
    >
      someDisplayName 
    </span>
    <span
      class="message"
    >
      someMessage
    </span>
  </div><div
    class="info"
  >
    <span
      class="created-text"
    >
      1st January um 12:00 AM
    </span>
  </div>
</app-comment>
`;
